### PR TITLE
[Enhancement] Support configuring HMS timeout param when creating hive/iceberg connector

### LINKED ISSUE #39593: SR should support configuring HMS timeout param when creating hive/iceberg connector
## Enhancement
It's strange to make hms timeout configuration static on FE side, as this timeout is a frequently used parameter, users may want to configure different timeout value based on different connector.

I want to make the hms timeout param configurable when creating connector. I added a new property hive.metastore.timeout. We can use it like this:

```
CREATE EXTERNAL CATALOG hiveconnector
PROPERTIES
(
"type" = "hive",
"hive.metastore.type" = "hive",
"hive.metastore.timeout" = "600",
"hive.metastore.uris" = "thrift://127.0.0.1:9083"
);
```

### PR DESCRIPTION
**Pull request by zhangbutao**
## Why I'm doing:

It's strange to make hms timeout configuration static on FE side, as this timeout is a frequently used parameter, users may want to configure different timeout value based on different connector.

I think it's good to configure the hms timeout when creating hive/icebeg connector.

## What I'm doing:
I want to make the hms timeout param configurable when creating connector. I added a new property `hive.metastore.timeout`. We can use it like this:
``` 
CREATE EXTERNAL CATALOG hiveconnector
PROPERTIES
(
"type" = "hive",
"hive.metastore.type" = "hive",
"hive.metastore.timeout" = "600",
"hive.metastore.uris" = "thrift://127.0.0.1:9083"
);
```

Fixes #39593

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by zhangbutao** [comment]
@stephen-shelby Do you have any suggestion? Thx.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/hive/IcebergHiveCatalog.java [hunk 1]
```diff
@@ -66,8 +67,10 @@ public class IcebergHiveCatalog implements IcebergCatalog {
     @VisibleForTesting
     public IcebergHiveCatalog(String name, Configuration conf, Map<String, String> properties) {
         this.conf = conf;
-        this.conf.set(MetastoreConf.ConfVars.CLIENT_SOCKET_TIMEOUT.getHiveName(),
-                String.valueOf(Config.hive_meta_store_timeout_s));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/hive/IcebergHiveCatalog.java:70]: why not use this configuration?

Comment (zhangbutao) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/hive/IcebergHiveCatalog.java:70]: If user does not set the new added propertiy `hive.metastore.timeout` when creating catalog, then will use this default value `Config.hive_meta_store_timeout_s`.

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java [hunk 2]
```diff
@@ -78,8 +79,10 @@ public static HiveMetaClient createHiveMetaClient(HdfsEnvironment env, Map<Strin
         if (properties.containsKey(HIVE_METASTORE_URIS)) {
             conf.set(MetastoreConf.ConfVars.THRIFT_URIS.getHiveName(), properties.get(HIVE_METASTORE_URIS));
         }
-        conf.set(MetastoreConf.ConfVars.CLIENT_SOCKET_TIMEOUT.getHiveName(),
-                String.valueOf(Config.hive_meta_store_timeout_s));
+        String hmsTimeout = properties.containsKey(HIVE_METASTORE_TIMEOUT)
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java:82]: Map.getOrDefault()

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**